js/index.js

```diff
@@ -0,0 +1,52 @@
+
+const getUserTrips = (userID, trips) => {
+    if (typeof userID !== 'number') {
+        return 'You must enter a valid ID, please.'
+    }
+    const userTrips = trips.trips.filter(trip => trip.userID === userID)
+    if (!userTrips.length) {
+        return 'You have no trips on the itinerary.'
+    }
+
+    const approvedTrips = userTrips.filter(trip => trip.status === 'approved')
+    const pendingTrips = userTrips.filter(trip => trip.status === 'pending')
+
+    if (!approvedTrips.length && pendingTrips.length) {
+        return 'All your trips are pending approval.'
+    }
+    return { approvedTrips, pendingTrips }
+}
+
+const getUserExpenditures = (userID, trips, destinations) => {
+    const userTrips = getUserTrips(userID, trips);
+    if (typeof userTrips === 'string') {
+        return userTrips;
+    }
+    const { approvedTrips } = userTrips;
+    if (!approvedTrips.length) {
+        return 'All your trips are pending approval.';
+    }
+
+    const expendituresByYear = {};
+
+    approvedTrips.forEach(trip => {
+        const tripYear = new Date(trip.date).getFullYear();
```

Comment:
I like that you're able to see expenditures by different years but since the project scope asks for the latest user year, I'd try to highlight that in the function itself. You can do a separate forEach to get the highest number year